Element84/filmdrop-ui
# Ba/feature/adding layer list

**Related Issue(s):**

- NA

**Proposed Changes:**

1. Adds config options for enabling and setting config for wms reference layers
2. Adds toggle-able reference layers list if config values are enabled and layers added
3. Updates readme with reference layers details

BONUS: Added config options to set map starting zoom and center

### To Test:

**Test layer list**
- set local config values for `Layer_List_Enabled: true` & `Layer_List_Services:` to match array example sent in DM. 
- load app
- confirm layer list toggle button renders in app
- confirm example layers get added to map
- click layer list toggle button
- confirm reference layer component renders
- confirm checkboxes match default visibilities set for layers in config
- confirm layer in list are rendered in same order they appear in config
- confirm layers on map are rendered in the same order as the layer list
- check and uncheck layers
- confirm layer order is maintained after toggling layers
- click search
- confirm result footprints render above reference layers
- click image footprint
- confirm image overlay renders
- confirm image overlay renders above reference layers 
- click mosaic toggle
- confirm mosaic renders above reference layers 
- with footprints & overlay showing toggle reference layers on and off
- confirm layer rendering order stays the same as reference list and reference layer group still renders below all other layers
- click layer list toggle button again
- confirm reference layer component closes
- confirm reference layers still persist on map

**test no layer list**
- remove the `Layer_List_Services` from the config
- load app
- confirm layer list toggle button does not render
- stop app
- `Layer_List_Services` back to config 
- set `Layer_List_Enabled: false`
- load app
- confirm layer list toggle button does not render

**test bonus**
- load app without `MAP_ZOOM` or `MAP_CENTER` set in config
- confirm app default to center of globe and low zoom level (3)
- stop app
- add `MAP_ZOOM: 8` to config
- load app 
- confirm map initializes to zoom level 8
- stop app
- add `MAP_CENTER: [-23.646732796934216, -46.66332945171046]` to config
- load app
- confirm map starting location matches matches `MAP_CENTER` value

**PR Checklist:**
